storm-core/src/jvm/storm/trident/spout/OpaquePartitionedTridentSpoutExecutor.java

```diff
@@ -162,9 +162,9 @@ public void commit(TransactionAttempt attempt) {
             
             Long txid = attempt.getTransactionId();
             Map<String, Object> metas = _cachedMetas.remove(txid);
-            for(String partitionId: metas.keySet()) {
-                Object meta = metas.get(partitionId);
-                _partitionStates.get(partitionId).rotatingState.overrideState(txid, meta);
+            for(Entry<String, Object> entry: metas.entrySet()) {
+                Object meta = entry.getValue();
```

Comment:
Unless the variable is created to make the code easier to read, this could be simplified to   _partitionStates.get(entry.getKey()).rotatingState.overrideState(txid,  entry.getValue());